float32 xhat
float32 yhat
float32 zhat1
float32 zhat2
float32 Vxhat
float32 Vyhat
float32 Vhat
float32 psihat